fl4health/clients/moon_client.py

```diff
@@ -95,7 +95,7 @@ def get_contrastive_loss(
         return self.ce_criterion(logits, labels)
 
     def update_after_train(self, local_steps: int, loss_dict: Dict[str, float]) -> None:
-        assert isinstance(self.model, MoonModel)
+        assert isinstance(self.model, SequentiallySplitModel)
```

Comment:
@sanaAyrml: We can make Moon models inherit from something else entirely if you think that makes sense. The only good thing about them sharing a base class with the FedRep and FedPer clients is that they could, theoretically still be used with a MoonClient.

Reply:
Sure. I'll do that instead!